pybamm/models/submodels/external_circuit/function_control_external_circuit.py

```diff
@@ -6,22 +6,41 @@
 
 
 class FunctionControl(BaseModel):
-    """External circuit with an arbitrary function."""
+    """
+    External circuit with an arbitrary function, implemented as a control on the current
+    either via an algebraic equation, or a differential equation.
+
+    Parameters
+    ----------
+    param : parameter class
+        The parameters to use for this submodel
+    external_circuit_function : callable
+        The function that controls the current
+    control : str, optional
+        The type of control to use. Must be one of 'algebraic' (default)
+        or 'differential'.
+    """
 
-    def __init__(self, param, external_circuit_function):
+    def __init__(self, param, external_circuit_function, control="algebraic"):
         super().__init__(param)
         self.external_circuit_function = external_circuit_function
+        self.control = control
 
     def get_fundamental_variables(self):
         param = self.param
         # Current is a variable
-        i_cell = pybamm.Variable("Total current density")
+        i_var = pybamm.Variable("Current density variable")
+        if self.control == "algebraic":
+            i_cell = i_var
+        elif self.control == "differential":
+            i_cell = pybamm.maximum(i_var, param.current_with_time)
```

Comment:
Might be useful to add a comment or point to a reference saying why you get `maximum` here